RangeEnhancedIterator_Generator.py

```diff
@@ -0,0 +1,90 @@
+class IteratorEnhancedRange:
+    
+    def __init__(self, *args):
+    #Arguments are: start (default = 0), end (no default, required argument), step (default = 1)
+        self.args = args
+        for arg in args:
+            if not (type(arg) == int or type(arg) == float):
+                raise Exception("Arguments must be numeric")
+        if len(args) == 1:
+            self.start = 0
+            self.end = self.args[0]
+            self.step = 1
+        elif len(args) == 2:
+            self.start = self.args[0]
+            self.end = self.args[1]
+            self.step = 1
+        elif len (args) == 3:
+            if self.args[2] == 0:
+                raise Exception("Unable to range with step = 0")
+            self.start = self.args[0]
+            self.end = self.args[1]
+            self.step = self.args[2]
+        else:
+            raise Exception("IteratorEnhancedRange requires at least one, but no more than three arguments")
+        self.startediter = False
```

Comment:
Could we get rid of this flag? It seems to be used only ones.

Reply:
We probably could by making the start point at (start-1) and then move to the main cycle, where +1 will return it to the start. But I like my version better as the flag clearly distinguishes whether the iteration has or has not started.

Reply:
Update: did all the things I wrote about in Generator's code, but still prefer to retain Iterator as it is.